### PR TITLE
add support for NFD-Master instance flag

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
Issue https://github.com/kubernetes-sigs/node-feature-discovery/issues/427 proposed a way to handle multiple NFD instances on a same cluster , this was addressed with https://github.com/kubernetes-sigs/node-feature-discovery/pull/431 patch 

this patch updated the operator to support multiple instances of NFD on the same cluster

THIS PATCH MUST BE REVIEWED AND MERGED AFTER #53

### CODE REVIEW

**Comment by ArangoGutierrez** [comment]
this PR buils on PR #53 so leaving it on 
/hold

File: config/crd/bases/nfd.kubernetes.io_nodefeaturediscoveries.yaml
Comment (marquiz) [inline config/crd/bases/nfd.kubernetes.io_nodefeaturediscoveries.yaml:8]: Unrelated, unnecessary?

File: config/rbac/role.yaml
Comment (marquiz) [inline config/rbac/role.yaml:319]: Makes sense but could this be addressed in a separate PR (or at least patch)?

File: controllers/nodefeaturediscovery_controls.go
Comment (marquiz) [inline controllers/nodefeaturediscovery_controls.go:320]: stupid question: can the instance flag change so that this code is unable to find it?

**Review by marquiz — CHANGES_REQUESTED** [review]

File: controllers/nodefeaturediscovery_controls.go
Comment (ArangoGutierrez) [reply controllers/nodefeaturediscovery_controls.go:320]: Good catch!

Comment (marquiz) [reply controllers/nodefeaturediscovery_controls.go:320]: Hmm, could we drop this fragile-looking logic altogether and just forcefully (re-)set the container command line?

E.g. so that `build/assets/master/0400_master_daemonset.yaml` would have simply `command: ["nfd-master"]` and we would set `obj.Spec.Template.Spec.Containers[0].Args` to `{"--port=12000"}` here and append `--instance` if necessary. Would make it a lot cleaner and more robust IMO. WDYT?

**Review by marquiz — CHANGES_REQUESTED** [review]

File: config/samples/nfd.kubernetes.io_v1_nodefeaturediscovery.yaml
Comment (ArangoGutierrez) [reply config/samples/nfd.kubernetes.io_v1_nodefeaturediscovery.yaml:127]: this patch was created/checkout on #53 git story , that's why I have it on hold. I will update once we merge the core-config patch

**Comment by ArangoGutierrez** [comment]
Now this baby needs rebase

File: controllers/nodefeaturediscovery_controls.go
Comment (ArangoGutierrez) [reply controllers/nodefeaturediscovery_controls.go:320]: agree, also TODO allow users to define the master port via CRD, WDYT? 
cc @zvonkok

**Comment by ArangoGutierrez** [comment]
This Patch now looks a lot better and more specific to it's intention 
ready for review @marquiz

File: build/assets/master/0400_master_daemonset.yaml
Comment (marquiz) [inline build/assets/master/0400_master_daemonset.yaml:33]: I think you could/should drop this from here and add it unconditionally in the controller.

File: controllers/nodefeaturediscovery_controls.go
Comment (marquiz) [reply controllers/nodefeaturediscovery_controls.go]: Yes of course. I'm not sure how useful that would be 🤔 Maybe it is

**Comment by ArangoGutierrez** [comment]
Will rebase on #63 to unhold

**Comment by ArangoGutierrez** [comment]
@marquiz this one is now ready to smash

File: controllers/nodefeaturediscovery_controls.go
Comment (marquiz) [inline controllers/nodefeaturediscovery_controls.go:325]: nit: I think it would be neater and more future-proof (if we add more flags) to use `append()` here

**Review by marquiz — APPROVED** [review]
I like this @ArangoGutierrez! Good work 😎 

/lgtm

**Pull request merged**
